python/cudf/cudf/core/frame.py

```diff
@@ -1248,14 +1251,26 @@ def replace(self, to_replace, replacement):
                         column=col,
                     )
 
-                    copy_data[name] = col.find_and_replace(
-                        col_to_replace, col_replacement, col_all_nan
-                    )
+                    # Index values to be replaced do not store in column
+                    # data but instead *_values*. Column accessor bypassed
+                    # to direct to Index.find_and_replace()
+                    if is_index:
+                        copy_data[name] = self.find_and_replace(
+                            to_replace=to_replace, replacement=replacement,
+                        )
+                    else:
+                        copy_data[name] = col.find_and_replace(
+                            col_to_replace, col_replacement, col_all_nan
+                        )
```

Comment:
Index still inherits from `Frame`: https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/core/index.py#L60 so I'm not following why you need to change the call to ignore `col_all_nan` here.